sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs

```diff
@@ -201,16 +209,16 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                         : _serviceClient.HealthStatus(new Guid(Id), _top, _skip, _showStats, cancellationToken);
 
                     _response = update.GetRawResponse();
+                    _status = update.Value.Status;
 
-                    if (update.Value.Status == JobStatus.Succeeded)
+                    if (update.Value.Status == TextAnalyticsOperationStatus.Succeeded)
                     {
                         // we need to first assign a vaue and then mark the operation as completed to avoid race conditions
                         _value = Transforms.ConvertToRecognizeHealthcareEntitiesResultCollection(update.Value.Results, _idToIndexMap);
-
                         NextLink = update.Value.NextLink;
                         _hasCompleted = true;
                     }
-                    else if (update.Value.Status == JobStatus.Failed)
+                    else if (update.Value.Status == TextAnalyticsOperationStatus.Failed || update.Value.Status == TextAnalyticsOperationStatus.Cancelled)
```

Comment:
if the operation is cancelled, does update.Value have content? and does it contain errors?
If yes, ignore the following :P.

If not, I liked what you had before on the `Value` property of creating a new `RequestFailedException` with no data available or something like that. But instead of creating the exception on `Value` create it here.
Let me know if that makes sense

Reply:
if the request is cancelled there is nothing returned in the body, whereas there can be failed cancelled call which can lead to 400 bad request if the job is completed

Reply:
What I am trying to say is that line  
```
 _requestFailedException = await ClientCommon.CreateExceptionForFailedOperationAsync(async, _diagnostics, _response, update.Value.Errors).ConfigureAwait(false);
```

Is not going to provide a useful message for the user. See how the service returns `errors:[]` which means that the `RequestFailedException` will not have information for the user.

```
"ResponseBody": {
        "jobId": "6957a3af-1f32-421c-a909-492cf6b72ac2",
        "lastUpdateDateTime": "2021-01-25T17:10:37Z",
        "createdDateTime": "2021-01-25T17:10:36Z",
        "expirationDateTime": "2021-01-26T17:10:36Z",
        "status": "cancelled",
        "errors": []
      }
```

Because of that, the code for when `update.Value.Status == TextAnalyticsOperationStatus.Cancelled` is true should change. It should contain:
- Create a new `RequestFailedException` with a useful message to the user about how the operation was cancelled and no results returned. ->similar to a message you had in the first iteration of your PR
- _hasCompleted = true;